text/2018-10-9-reduce-one-copy-in-grpc-send.md

```diff
@@ -68,42 +63,17 @@ The reason why still one copy:
 + C-side uses a ref-count mechanism to manage the lifetime of objects
 + We either need to extend the Rust objects' lifetime or do a copy (see alternatives section)
 
-# Alternatives
-
-## Reduce re-allocations
-
-Since we're copying data into a `Vec<u8>`, there must be a lot of re-allocations when we continuously add data into it.
-I see protobuf can compute the deserialized data size, so I added a `reserve` before copying data.
-
-### Result
-
-This does not work as good as the accepted solution and we cannot assume that other users can compute the data size too, so it's given up.
-
-## Dirty hack
-
-This idea prevents the second copy which originally happens on C-side.
-Since we cannot disbale Rust's deallocation, we can avoid using Rust objects at the very beginning.
-
-I created a C++ vector. We add data to it, and take the raw data pointer away.
-Now we have a pointer that points to a piece of memory that we only have to worry about when to deallocate it.
-This can be easily done with the ref-count mechanism.
-
-### Result
+# Unresolved questions
 
-Superseded by the use of `GrpcByteBuffer`.
+Actuallty I've tried to extend Rust objects' lifetime.
```

Comment:
Better to mention why it did not work out. Also 80 characters per line is preferred.